oci/layout/oci_transport.go

```diff
@@ -186,14 +188,28 @@ func (ref ociReference) NewImage(ctx *types.SystemContext) (types.Image, error)
 	return image.FromSource(src)
 }
 
-func (ref ociReference) getManifestDescriptor() (imgspecv1.Descriptor, error) {
+func (ref ociReference) getIndex() (imgspecv1.Index, error) {
+	index := imgspecv1.Index{
+		Versioned: imgspec.Versioned{
+			SchemaVersion: 2,
+		},
+	}
+
 	indexJSON, err := os.Open(ref.indexPath())
+	defer indexJSON.Close()
 	if err != nil {
-		return imgspecv1.Descriptor{}, err
+		return index, err
```

Comment:
I don’t think this should silently swallow I/O errors and return an empty index in that case as if everything was just fine and any pre-existing images inexplicably disappeared.

Reply:
Ah, it is actually returning a non-`nil` error value, I didn’t notice, sorry.

Anyway, yes, when the only caller ignores the index if `err != nil`, this function might as well return `nil, err` instead of pretending that returning `index` matters.

Reply:
Still outstanding.

Reply:
Ok, I'll return a *imgspecv1.Index. Not sure whether this is so much better though.

Reply:
(Returning an `imgspecv1.Index{}` would be fine as well; the point is that the common case is that all other return values are ignored on a non-`nil` `error` return, and this makes a misleading impression that the return value matters.)